main.py

```diff
@@ -95,29 +95,48 @@ async def poll_events():
 async def update_events(guild: discord.Guild, events, discord_events):
     updated = 0
     inserted = 0
+    duplicates_removed = 0
     # Prints the start and name of the next 10 events
     for event in events:
         start = event['start'].get('dateTime', event['start'].get('date'))
-        description = markdownify(event['description'])[:min(1000, len(event['description']))].strip()
+        # Skip all-day events
+        if datetime.fromisoformat(start).tzinfo is None:
+            continue
+
+        # This is mostly for debugging because all events *should* have a description, but some of mine don't
+        description = None
+        if "description" in event:
+            description = markdownify(event['description'])[:min(1000, len(event['description']))].strip()
+        else:
+            description = ""
+
+
         matching_discord_events = [e for e in discord_events if
                                    e.start_time - datetime.fromisoformat(start) == timedelta(0)
                                    and e.name.strip() == event['summary'].strip()]
         if len(matching_discord_events) > 0:
+            # Remove duplicate events
+            if len(matching_discord_events) > 1:
+                for i in range(1, len(matching_discord_events)):
+                    e = matching_discord_events[i]
+                    await e.delete()
```

Comment:
We don't want to delete duplicates, we need to prevent posting them in the first place. If we delete duplicates, there is no guarantee it will be the new/dup one and not the original/RSVPed to event